x/rollup/tx/helpers/ante.go

```diff
@@ -3,22 +3,57 @@ package helpers
 import (
 	"fmt"
 
+	storetypes "cosmossdk.io/store/types"
+	txsigning "cosmossdk.io/x/tx/signing"
 	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
 	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	rollupkeeper "github.com/polymerdao/monomer/x/rollup/keeper"
 	rolluptx "github.com/polymerdao/monomer/x/rollup/tx"
 )
 
 type AnteHandler struct {
-	authAnte sdktypes.AnteHandler
+	authAnteHandler   sdktypes.AnteHandler
+	l1DataAnteHandler sdktypes.AnteHandler
 }
 
-func NewAnteHandler(options authante.HandlerOptions) (*AnteHandler, error) { //nolint:gocritic // hugeParam
-	authAnteHandler, err := authante.NewAnteHandler(options)
+// HandlerOptions are the options required for constructing the AnteHandler used for non-deposit txs.
+// The options are the same as those required for constructing the default x/auth AnteHandler with the
+// addition of the RollupKeeper.
+type HandlerOptions struct {
+	AccountKeeper          authante.AccountKeeper
+	BankKeeper             authtypes.BankKeeper
+	RollupKeeper           *rollupkeeper.Keeper
+	ExtensionOptionChecker authante.ExtensionOptionChecker
+	FeegrantKeeper         authante.FeegrantKeeper
+	SignModeHandler        *txsigning.HandlerMap
+	SigGasConsumer         func(meter storetypes.GasMeter, sig signing.SignatureV2, params authtypes.Params) error
+	TxFeeChecker           authante.TxFeeChecker
+}
```

Comment:
Rather than modifying the options. Can we just pass in the rollup keeper as a separate argument?